devito/ops/types.py

```diff
@@ -9,6 +9,14 @@
 
 class Array(basic.Array):
 
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._allocation = kwargs.get('allocation')
```

Comment:
do you really need an extra attribute? can you not reuse `scope` ?

Reply:
In your comment on slack, I understood that I should not use scope:

> I’m not particularly fond of the _scope attribute for reasons that we can discuss another day, so if we managed to find a way not to touch it, I’d prefer

As we have:
```
self._scope = kwargs.get('scope', 'heap')
assert self._scope in ['heap', 'stack']
```
I would need to change scope

Reply:
I'll change that `assert` and overwrite _mem_external

Reply:
> In your comment on slack, I understood that I should not use scope

right, but if the alternative is to add a further attribute, then we can just use scope I'd say?

Reply:
ahh. Ok! Changed.